pydantic_ai_slim/pydantic_ai/durable_exec/temporal/__init__.py

```diff
@@ -91,6 +93,22 @@ def __init__(self):
             workflow_failure_exception_types=[UserError, PydanticUserError],
         )
 
+    def configure_worker(self, config: WorkerConfig) -> WorkerConfig:
+        config = super().configure_worker(config)
+
+        workflows = list(config.get('workflows', []))  # type: ignore[reportUnknownMemberType]
+        activities = list(config.get('activities', []))  # type: ignore[reportUnknownMemberType]
+
+        for workflow_class in workflows:  # type: ignore[reportUnknownMemberType]
+            if issubclass(workflow_class, PydanticAIWorkflow):
+                agents = workflow_class.__pydantic_ai_agents__
```

Comment:
Note this line that I added to the issue description, possibly after you started working on this:

> But for convenience, I don't want to _require_ the workflow class to subclass that, so we should also do a `getattr(workflow_cls, '__pydantic_ai_agents__', None)` and check the types at runtime.